client/rest/src/test/java/org/elasticsearch/client/RestClientMultipleHostsIntegTests.java

```diff
@@ -94,9 +98,34 @@ private static HttpServer createHttpServer() throws Exception {
         for (int statusCode : getAllStatusCodes()) {
             httpServer.createContext(pathPrefix + "/" + statusCode, new ResponseHandler(statusCode));
         }
+        waitForCancelHandler = new WaitForCancelHandler();
+        httpServer.createContext(pathPrefix + "/wait", waitForCancelHandler);
         return httpServer;
     }
 
+    private static class WaitForCancelHandler implements HttpHandler {
+        private CountDownLatch cancelHandlerLatch;
```

Comment:
this can be `final`

Reply:
it gets reset before each request in testCancelAsyncRequests, so it can't be final I think